components/logins/android/src/test/java/mozilla/appservices/logins/DatabaseLoginsStorageTest.kt

```diff
@@ -21,5 +23,36 @@ class DatabaseLoginsStorageTest: LoginsStorageTest() {
         return DatabaseLoginsStorage(dbPath = dbPath.absolutePath)
     }
 
+    @Test
+    fun testUnlockHex() {
+        val store = createTestStore()
+        val key = "0123456789abcdef"
+        // This is a little awkward because kotlin/java Byte is signed, and so the literals
+        // above 128 (above 0x80) can't be part of a `listOf<Byte>()` (there's UByte, but it's
+        // both experimental and very unclear that JNA would do anything sane with it).
+        val keyBytes = listOf(0x01, 0x23, 0x45, 0x67, 0x89, 0xab, 0xcd, 0xef)
+                .map { it.toByte() }
+                .toByteArray()
+
+        store.unlock(keyBytes)
+
+        store.add(ServerPassword(
+                id = "aaaaaaaaaaaa",
+                hostname = "https://www.example.com",
+                httpRealm = "Something",
+                username = "Foobar2000",
+                password = "hunter2",
+                usernameField = "users_name",
+                passwordField = "users_password"
+        ))
+
+        store.lock()
+        // Ensure that it's equivalent to encrypting with the hex encoded string.
+        store.unlock(key)
```

Comment:
Can you assert that `unlock(badkey)` fails, and that `ensuredUnlocked()` succeeds both when the store is locked and the store is unlocked?

Reply:
This is covered elsewhere, but only for the string overload, so ok.